os/20211014-s390x-secure-execution.md

```diff
@@ -0,0 +1,190 @@
+
+## Firstboot
+
+During guest provisioning new hostkey(s) is/are fetched and used for generation of `sd-boot`. User should specify them within ignition config with pre-defined name template: `/etc/se-hostkeys/ibm-z-hostkey-N` where `N` is a key number, several hostkeys could be used:
+
+```yaml
+storage:
+  files:
+    -
+      path: /etc/se-hostkeys/ibm-z-hostkey-1
+      overwrite: true
+      contents:
+        source: https://url.com/key1
+```
+
+System would also set a new random LUKS keys for both `boot` and `root` filesystems, so each node has its own LUKS keys.
+
+## Lifetime
+
+Secure Execution is enabled by 2 tools provided by IBM's s390-utils package:
+- `genprotimg` - tool to generate `sd-boot`
+- `zipl` - tool to install new bootloader
+
+Despite `zipl` is well known and used by CoreOS, it couldn't be executed same way as before. When system runs in Secure Execution, `ostree` and `rdcore` automatically detect that and perform all required steps.
+That means, that user won't notice any difference, both mentioned components would automatically regenerate `sd-boot` image and
+install it with `zipl` whenever required.
+
+# Implementation steps
+
+## Step 1: CoreOS with existing patches to `libostree` could be configured to perform in Secure Execution mode.
+
+Customer's public hostkey(s) should be placed into `/etc/se-hostkeys/` using name templates `ibm-z-hostkey-N`. Secure Execution is enabled by hands following https://www.ibm.com/docs/en/linux-on-systems?topic=linux-introduction.
+
+## Step 2: New `qemu-secex.qcow2` build artifact for CoreOS with enabled Secure Execution.
+
+`coreos-xyz-qemu-secex.qcow2` image with LUKS encrypted `/boot` and `rootfs` is available for download. This is done by using new `buildextend-secex` `coreos-assembler` command, which:
+ - adds new `/se` partition for `sd-boot`
+ - encrypts `/boot` and `/` with random LUKS keys
+ - stores LUKS keys and `crypttab` config to `/etc`
+ - enables a rough equivalent of the effect of TPM-bound LUKS on z15, by appennding the LUKS keys and `crypttab` to the `initramfs`, which is encrypted as part of `sd-boot`. Original `initramfs` under `/boot` is non-modified
+ - generates `/se/sd-boot` encrypted by universal public hostkey (valid for all Z15 and newer machines)
+ - `zipl`es `/se/sd-boot`, so `qemu-secex.qcow2` image could only be run with Secure Execution
+
+Customer should add his public hostkeys(s) into `ignition` config.
+> **Note:** We cannot use LUKS `integrity` option, because it doesn's support resizing of partitions, so `rootfs` cannot be growned during `firstboot`.
+
+## Step 3: During firstboot `/boot` and `rootfs` are reencrypted with randomly generated LUKS keys and `integrity` option, which is deffered from Step 2
+
+During firstboot the LUKS keys will be changed to a randomly generated ones. The idea behind this is to set on each host the LUKS keys to a unique keys during firstboot. This should prevent reading the keys from the provided pre-build image.
```

Comment:
Add something like: "Also, LUKS integrity protection will be enabled both in the pre-built image and after re-encryption."